ts/features/uaDonations/components/__test__/UaDonationsBanner.test.tsx

```diff
@@ -0,0 +1,223 @@
+// import React from "react";
+import { NavigationParams } from "react-navigation";
+import { createStore, Store } from "redux";
+import { MockStore } from "redux-mock-store";
+import ROUTES from "../../../../navigation/routes";
+import { applicationChangeState } from "../../../../store/actions/application";
+import { backendStatusLoadSuccess } from "../../../../store/actions/backendStatus";
+import { appReducer } from "../../../../store/reducers";
+import { baseRawBackendStatus } from "../../../../store/reducers/__mock__/backendStatus";
+import { GlobalState } from "../../../../store/reducers/types";
+import { getFullLocale } from "../../../../utils/locale";
+import { renderScreenFakeNavRedux } from "../../../../utils/testWrapper";
+import { UaDonationsBanner } from "../UaDonationsBanner";
+
+jest.mock("../../../../config", () => ({ uaDonationsEnabled: true }));
+
+describe("UaDonationsBanner", () => {
+  jest.useFakeTimers();
+
+  describe("When the store is in default state and BACKEND_STATUS_LOAD_SUCCESS has not been dispatched", () => {
+    it("Should not render the UaDonationsBanner", () => {
+      const globalState = appReducer(
+        undefined,
+        applicationChangeState("active")
+      );
+      const store = createStore(appReducer, globalState as any);
+      const result = renderComponent(store);
+
+      expect(result.component.queryByTestId("UaDonationsBanner")).toBeNull();
+    });
+  });
+
+  describe("When a backendStatusLoadSuccess is received", () => {
+    describe("And uaDonations.enabled === false, uaDonations.banner.visible === false", () => {
```

Comment:
Do you think can be an improvement? 
I group all the `Should not render the UaDonationsBanner` cases.

The result is the same, so feel free to ignore it.
```

describe.each`
      uaDonationsEnabled | uaDonationsBannerVisible | uaDonationsBannerDescription
      ${false}           | ${false}                 | ${baseRawBackendStatus.config.uaDonations.banner.description}
      ${true}            | ${false}                 | ${baseRawBackendStatus.config.uaDonations.banner.description}
      ${false}           | ${true}                  | ${baseRawBackendStatus.config.uaDonations.banner.description}
      ${true}            | ${true}                  | ${{ "it-IT": "", "en-EN": "" }}
    `(
      "And uaDonations.enabled === $uaDonationsEnabled, uaDonations.banner.visible === $uaDonationsBannerVisible and uaDonations.banner.description === $uaDonationsBannerDescription",
      ({
        uaDonationsEnabled,
        uaDonationsBannerVisible,
        uaDonationsBannerDescription
      }) => {
        const globalState = appReducer(
          undefined,
          applicationChangeState("active")
        );
        const store = createStore(appReducer, globalState as any);
        store.dispatch(
          backendStatusLoadSuccess({
            ...baseRawBackendStatus,
            config: {
              ...baseRawBackendStatus.config,
              uaDonations: {
                ...baseRawBackendStatus.config.uaDonations,
                enabled: uaDonationsEnabled,
                banner: {
                  ...baseRawBackendStatus.config.uaDonations.banner,
                  visible: uaDonationsBannerVisible,
                  description: uaDonationsBannerDescription
                }
              }
            }
          })
        );
        it("Should not render the UaDonationsBanner", () => {
          const result = renderComponent(store);

          expect(
            result.component.queryByTestId("UaDonationsBanner")
          ).toBeNull();
        });
      }
    );
```